js/style/style.js

```diff
@@ -18,12 +18,13 @@ var QueryFeatures = require('../source/query_features');
 var SourceCache = require('../source/source_cache');
 var styleSpec = require('./style_spec');
 var StyleFunction = require('./style_function');
+var getWorkerPool = require('../global_worker_pool');
```

Comment:
@lucaswoj I changed to the lazy-lookup approach to allow for using `mapboxgl.workerCount`.  

However, it occurs to me that there's a reason to keep the singleton exposed as a separate module: once this PR goes in, we could adjust https://github.com/mapbox/mapbox-gl-js/pull/2982 so that WorkerSources are registered directly in `Source#addType`.  The most straightforward way to do that, IMO, would be to allow `source.js` to pull in a reference to the global WorkerPool instance.

^ If the value of doing that isn't clear from that description, I could also just make this a file-scoped variable as you're suggesting now, and re-extract it as part of that PR.